src/library_fetchfs.js

```diff
@@ -5,11 +5,20 @@
  */
 
 addToLibrary({
-  $FETCHFS__deps: ['$stringToUTF8OnStack', 'wasmfs_create_fetch_backend'],
+  $FETCHFS__deps: ['$stringToUTF8OnStack', 'wasmfs_create_fetch_backend', 'wasmfs_fetch_create_manifest', 'wasmfs_fetch_add_to_manifest'],
   $FETCHFS: {
     createBackend(opts) {
-      return _wasmfs_create_fetch_backend(stringToUTF8OnStack(opts.base_url));
-    }
+      return withStackSave(() => {
+        var manifest = 0;
+        if (opts['manifest']) {
+          manifest = wasmfs_fetch_create_manifest();
```

Comment:
This should require a `_` prefix, shouldn't it? New line 20 shows such a prefix being used for another function. Does this work without the prefix?

Reply:
Good question. Is there a good way to write unit tests for the JS side of interop? Say, a test that used JS to create a fetchfs filesystem and then run a C program that checks for he presence/contents of files in the filesystem?  I’d be happy to add a test case if you can point me to an example of a test that isn’t just C code.

Reply:
You can embed JS code into the C code using `EM_ASM` etc. See e.g.

https://github.com/emscripten-core/emscripten/blob/5643c2694ce08c4cfcada3529b8271ae5721e56e/test/wasmfs/wasmfs_getdents.c#L100-L108

Reply:
Ok that’s helpful, and would at least tell us if the createBackend code path works.  I was wondering if we could have a test where JS code in a pre run hook can successfully set up a filesystem—is there something like em_asm that runs before main but after global initializers?